### ISSUE TITLE
Packer 1.9.2 is unable to find a custom plugin on the current working directory

### ISSUE THREAD

**Issue by nywilken** [hashicorp/packer#12543]
Packer 1.9.2 seems that it is unable to find a custom plugin on the current working directory where packer init runs

Issue reported on [Packer forum](https://discuss.hashicorp.com/t/packer-1-9-2-seems-that-it-is-unable-to-find-a-custom-plugin-on-the-current-working-directory-where-packer-init-runs/56523/1) by [smaruyama](https://discuss.hashicorp.com/u/smaruyama/summary)

----
Hello,

I found that packer version 1.9.2 is unable to find a custom plugin on the current working directory where I run packer init

The following is the error log.
```
$ ls main.pkr.hcl packer-plugin-docker
main.pkr.hcl		packer-plugin-docker
$
$ packer version
Packer v1.9.2
$ packer init .
$ packer build -only='build.docker.amd64' -except='pp-*' .
Warning: Bundled plugins used

This template relies on the use of plugins bundled into the Packer binary.
The practice of bundling external plugins into Packer will be removed in an
upcoming version.

To remove this warning, add the following section to your template:

packer {
  required_plugins {
    amazon = {
      source  = "github.com/hashicorp/amazon"
      version = "~> 1"
    }
    docker = {
      source  = "github.com/hashicorp/docker"
      version = "~> 1"
    }
    vagrant = {
      source  = "github.com/hashicorp/vagrant"
      version = "~> 1"
    }
    ansible = {
      source  = "github.com/hashicorp/ansible"
      version = "~> 1"
    }
  }
}

Then run 'packer init' to manage installation of the plugins


build.docker.amd64: output will be in this color.

==> build.docker.amd64: Creating a temporary directory for sharing data...
==> build.docker.amd64: Pulling Docker image: amd64/amazonlinux:2
    build.docker.amd64: 2: Pulling from amd64/amazonlinux
    build.docker.amd64: Digest: sha256:4a3e160c969a06fed62fa091d6cf7e20387b40f330bd0393118cb17bd3abf433
    build.docker.amd64: Status: Image is up to date for amd64/amazonlinux:2
    build.docker.amd64: docker.io/amd64/amazonlinux:2
    build.docker.amd64: What's Next?
    build.docker.amd64: View summary of image vulnerabilities and recommendations → docker scout quickview amd64/amazonlinux:2
==> build.docker.amd64: Starting docker container...
    build.docker.amd64: Run command: docker run --privileged --platform linux/amd64 -v /var/run/secrets/:/var/run/secrets/ -v ~/.aws:/root/.aws -v /Users/{{ username }}/.packer.d/tmp3451986301:/packer-files -d -i -t --entrypoint=/bin/sh -- amd64/amazonlinux:2
==> build.docker.amd64: Error running container: Docker exited with a non-zero exit status.
==> build.docker.amd64: Stderr: docker: Error response from daemon: create ~/.aws: "~/.aws" includes invalid characters for a local volume name, only "[a-zA-Z0-9][a-zA-Z0-9_.-]" are allowed. If you intended to pass a host directory, use absolute path.
==> build.docker.amd64: See 'docker run --help'.
==> build.docker.amd64:
Build 'build.docker.amd64' errored after 2 seconds 373 milliseconds: Error running container: Docker exited with a non-zero exit status.
Stderr: docker: Error response from daemon: create ~/.aws: "~/.aws" includes invalid characters for a local volume name, only "[a-zA-Z0-9][a-zA-Z0-9_.-]" are allowed. If you intended to pass a host directory, use absolute path.
See 'docker run --help'.


==> Wait completed after 2 seconds 373 milliseconds

==> Some builds didn't complete successfully and had errors:
--> build.docker.amd64: Error running container: Docker exited with a non-zero exit status.
Stderr: docker: Error response from daemon: create ~/.aws: "~/.aws" includes invalid characters for a local volume name, only "[a-zA-Z0-9][a-zA-Z0-9_.-]" are allowed. If you intended to pass a host directory, use absolute path.
See 'docker run --help'.


==> Builds finished but no artifacts were created.
```

But if I export the PACKER_PLUGIN_PATH variable, packer build succeeds.

```
$ export PACKER_PLUGIN_PATH="$PWD"
$ packer build -only='build.docker.amd64' -except='pp-*' .
Warning: Bundled plugins used

This template relies on the use of plugins bundled into the Packer binary.
The practice of bundling external plugins into Packer will be removed in an
upcoming version.

To remove this warning, add the following section to your template:

packer {
  required_plugins {
    docker = {
      source  = "github.com/hashicorp/docker"
      version = "~> 1"
    }
    vagrant = {
      source  = "github.com/hashicorp/vagrant"
      version = "~> 1"
    }
    ansible = {
      source  = "github.com/hashicorp/ansible"
      version = "~> 1"
    }
    amazon = {
      source  = "github.com/hashicorp/amazon"
      version = "~> 1"
    }
  }
}

```
Then run 'packer init' to manage installation of the plugins

```
build.docker.amd64: output will be in this color.

==> build.docker.amd64: Creating a temporary directory for sharing data...
==> build.docker.amd64: Pulling Docker image: amd64/amazonlinux:2
    build.docker.amd64: 2: Pulling from amd64/amazonlinux
    build.docker.amd64: Digest: sha256:4a3e160c969a06fed62fa091d6cf7e20387b40f330bd0393118cb17bd3abf433
    build.docker.amd64: Status: Image is up to date for amd64/amazonlinux:2
    build.docker.amd64: docker.io/amd64/amazonlinux:2
    build.docker.amd64: What's Next?
    build.docker.amd64: View summary of image vulnerabilities and recommendations → docker scout quickview amd64/amazonlinux:2
==> build.docker.amd64: Starting docker container...
    build.docker.amd64: Run command: docker run --privileged --platform linux/amd64 -v /var/run/secrets/:/var/run/secrets/ -v /Users/{{ username }}/.aws:/root/.aws -v /Users/{{ username }}/.packer.d/tmp3353669033:/packer-files -d -i -t --entrypoint=/bin/sh -- amd64/amazonlinux:2
    build.docker.amd64: Container ID: ac6891cf97af1a2125663d31b5c168e108ef46a51f3825256aa120c9428ff820
==> build.docker.amd64: Using docker communicator to connect: 192.168.1.2
==> build.docker.amd64: Provisioning with shell script: files/pre_build.sh
==> build.docker.amd64: + echo y

... ...
... ommitted for simplicity
... ...

==> Builds finished but no artifacts were created.
The plugin document says:

```
Packer uses the following process for loading the correct plugin:

1. All directories under the PACKER_PLUGIN_PATH environment variable, if PACKER_PLUGIN_PATH is set. The PACKER_PLUGIN_PATH takes precedences over all other plugin directories; no other directories will be checked.
1. The directory where packer is installed, or the executable directory.
1. The current working directory, where packer build is being invoked. (“.”)
1. The PACKER_CONFIG_DIR/plugins directory. PACKER_CONFIG_DIR refers to Packer’s config directory, if it could be found.

The rule No.3 seems not working.
I would like to hear opinions from the community.
Thanks.

**Comment by nywilken** [comment]
This issue has been synced to JIRA for planning.
  
JIRA ID: [HPR-1281](https://hashicorp.atlassian.net/browse/HPR-1281)

**Comment by nywilken** [comment]
Hello :wave: thanks again for reporting, this should be fixed by #12544. There will be test binaries made available in the latest [nightly](https://github.com/hashicorp/packer/releases/tag/nightly) for testing prior to the next release. If you would like to test the fix and report back that would be greatly appreciated.
